src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -187,13 +189,16 @@ public Version getVersion() {
   }
 
   private boolean indexExists(String index) {
-    Action action = new IndicesExists.Builder(index).build();
-    try {
-      JestResult result = client.execute(action);
-      return result.isSucceeded();
-    } catch (IOException e) {
-      throw new ConnectException(e);
+    if (!indexCache.contains(index)) {
+      Action action = new IndicesExists.Builder(index).build();
+      try {
+        JestResult result = client.execute(action);
+        return result.isSucceeded();
+      } catch (IOException e) {
+        throw new ConnectException(e);
+      }
```

Comment:
nit: I'd write this code as:

```
if (indexCache.contains(index)) {
  return true;
} 

Action action = .... // the old code continues from here.
```

It's easier to read, and makes the changes look more succinct.